network-config-analyzer/CalicoPolicyYamlParser.py

```diff
@@ -580,17 +581,23 @@ def _get_selected_peers(self, policy_spec, is_profile, policy_name):
         :return: set of selected peers
         :rtype: PeerSet
         """
+        peers = self.peer_container.get_all_peers_group()  # default of selector/ ns_selector is all()
         pod_selector = policy_spec.get('selector')
         if pod_selector:
             if is_profile:
                 self.syntax_error('selector is not allowed in the spec of a Profile', policy_spec)
-            return self._parse_label_selector(pod_selector, policy_spec, self.namespace)
+            peers &= self._parse_label_selector(pod_selector, policy_spec, self.namespace)
         else:
             if is_profile:
                 return self.peer_container.get_profile_pods(policy_name, True)
             else:
                 return self.peer_container.get_namespace_pods(self.namespace)
 
+        ns_selector = policy_spec.get('namespaceSelector')
+        if ns_selector:
+            peers &= self._parse_label_selector(ns_selector, policy_spec, namespace_selector=True)
+        return peers
+
```

Comment:
Is it possible to have a global network policy with a namespace selector and  without a selector ? if yes, it seems this current version may miss this.

Reply:
I couldn't understand if this is possible from their docs.
from the "old" code, I concluded that before the namespace selector , a GlobalNetworkpolicy had to contain a selector. (all the cases under `else` doesn't match for it being a global networkPolicy (only profile or networkpolicy with a namespace)  and assumed the selector is still a must field.
I will try to test that on live cluster and fix if needed

Reply:
ok.
About the "cases under `else `doesn't match for it being a global networkPolicy" - I think it does match, because `self.namespace` can be `None ` for a global np.